### PR TITLE
Add different test type examples

### PR DESCRIPTION
**Pull request by lukaville**

### CODE REVIEW

File: android/app-example/src/androidTest/java/com/badoo/ribs/example/rib/menu/MenuTest.kt [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import com.badoo.ribs.example.rib.menu.element.MenuElement
+import com.jakewharton.rxrelay2.PublishRelay
+import io.reactivex.Observable
+import io.reactivex.ObservableSource
+import io.reactivex.functions.Consumer
+import io.reactivex.observers.TestObserver
+import org.junit.Rule
+import org.junit.Test
+
+class MenuTest {
```
Comment (lukaville) [inline android/app-example/src/androidTest/java/com/badoo/ribs/example/rib/menu/MenuTest.kt]: UI Android test that checks menu rib in isolation

File: android/app-example/src/androidTest/java/com/badoo/ribs/example/rib/menu/MenuViewTest.kt [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import com.badoo.ribs.example.app.OtherActivity
+import com.badoo.ribs.example.rib.menu.element.MenuElement
+import com.badoo.ribs.example.R
+import io.reactivex.ObservableSource
+import io.reactivex.observers.TestObserver
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+class MenuViewTest {
```
Comment (lukaville) [inline android/app-example/src/androidTest/java/com/badoo/ribs/example/rib/menu/MenuViewTest.kt]: UI Android test that check menu view in isolation

File: android/app-example/src/test/java/com/badoo/ribs/example/rib/menu/MenuRibTest.kt [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+import com.jakewharton.rxrelay2.Relay
+import com.nhaarman.mockitokotlin2.doReturn
+import com.nhaarman.mockitokotlin2.mock
+import io.reactivex.ObservableSource
+import io.reactivex.functions.Consumer
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Test
+
+class MenuRibTest {
```
Comment (lukaville) [inline android/app-example/src/test/java/com/badoo/ribs/example/rib/menu/MenuRibTest.kt]: local integration test that checks menu RIB without a real view

File: android/app-example/src/test/java/com/badoo/ribs/example/rib/menu/feature/MenuFeatureTest.kt [hunk 4]
```diff
@@ -0,0 +1,24 @@
+package com.badoo.ribs.example.rib.menu.feature
+
+import com.badoo.ribs.example.rib.menu.Menu
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+
+class MenuFeatureTest {
```
Comment (lukaville) [inline android/app-example/src/test/java/com/badoo/ribs/example/rib/menu/feature/MenuFeatureTest.kt]: feature unit tests

File: android/app-example/src/test/java/com/badoo/ribs/example/rib/switcher/SwitcherRouterTest.kt [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+import com.badoo.ribs.example.rib.switcher.dialog.DialogToTestOverlay
+import com.badoo.ribs.example.rib.util.TestNode
+import com.badoo.ribs.example.rib.util.subscribeOnTestObserver
+import com.nhaarman.mockitokotlin2.doReturn
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.verify
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+
+class SwitcherRouterTest {
```
Comment (lukaville) [inline android/app-example/src/test/java/com/badoo/ribs/example/rib/switcher/SwitcherRouterTest.kt]: router local test

File: android/app-example/src/test/java/com/badoo/ribs/example/rib/test/MainScreenTest.kt [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+import io.reactivex.ObservableSource
+import io.reactivex.functions.Consumer
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class MainScreenTest {
```
Comment (lukaville) [inline android/app-example/src/test/java/com/badoo/ribs/example/rib/test/MainScreenTest.kt]: local integration test of RIB subtree without real views

File: android/app-example/src/androidTest/java/com/badoo/ribs/example/rib/menu/MenuTest.kt [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    private val menuOutput = PublishRelay.create<Menu.Output>()
+
+    @Test
+    fun initialState_noSelectedElements() {
+        menu.assertNothingSelected()
+    }
+
+    @Test
+    fun selectItemInput_selectsItem() {
+        acceptInput(Menu.Input.SelectMenuItem(Menu.MenuItem.HelloWorld))
```
Comment (zsoltk) [inline android/app-example/src/androidTest/java/com/badoo/ribs/example/rib/menu/MenuTest.kt]: Some static imports would be nice.

File: android/app-example/src/test/java/com/badoo/ribs/example/rib/menu/MenuRibTest.kt [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+        rib.attachToView(mock())
+        rib.onStart()
+        rib.onResume()
+    }
+
+    @Test
+    fun selectItemInput_selectsItem() {
+        val viewModelObserver = menuView.viewModel.subscribeOnTestObserver()
+
+        menuInput.accept(Menu.Input.SelectMenuItem(Menu.MenuItem.HelloWorld))
```
Comment (Nublo) [inline android/app-example/src/test/java/com/badoo/ribs/example/rib/menu/MenuRibTest.kt]: static imports?

File: android/app-example/src/test/java/com/badoo/ribs/example/rib/menu/feature/MenuFeatureTest.kt [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+        val feature = MenuFeature()
+
+        assertThat(feature.state.selected).isNull()
+    }
+
+    @Test
+    fun `select menu item wish - updates state with the selected item`() {
+        val feature = MenuFeature()
+
+        feature.accept(Menu.Input.SelectMenuItem(Menu.MenuItem.HelloWorld))
```
Comment (Nublo) [inline android/app-example/src/test/java/com/badoo/ribs/example/rib/menu/feature/MenuFeatureTest.kt]: static imports?

File: android/app-example/src/test/java/com/badoo/ribs/example/rib/test/MainScreenTest.kt [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+        rootRib.onResume()
+
+        dependencies
+            .activityStarter
+            .stubResponse(component<OtherActivity>(), Instrumentation.ActivityResult(RESULT_OK, null))
+    }
+
+    @Test
+    fun openHelloSectionAndClickButton_launchesActivity() {
+        menuView.uiEvents.accept(MenuView.Event.Select(Menu.MenuItem.HelloWorld))
```
Comment (Nublo) [inline android/app-example/src/test/java/com/badoo/ribs/example/rib/test/MainScreenTest.kt]: static imports?

**Pull request merged**